framework/src/timeintegrators/CentralDifference.C

```diff
@@ -114,3 +114,30 @@ CentralDifference::computeTimeDerivatives()
     u_dot_factor.close();
   }
 }
+
+template <typename T, typename T2, typename T3, typename T4>
+void
+CentralDifference::computeTimeDerivativeHelper(T & u_dot,
+                                               T2 & u_dotdot,
+                                               const T3 & u_old,
+                                               const T4 & u_older) const
+{
+  // computing first derivative
+  // using the Central Difference method
+  // u_dot_old = (first_term - second_term) / 2 / dt
+  //       first_term = u
+  //      second_term = u_older
+  u_dot -= u_older; // 'older than older' solution
+  u_dot *= 1.0 / (2.0 * _dt);
+
+  // computing second derivative
+  // using the Central Difference method
+  // u_dotdot_old = (first_term - second_term + third_term) / dt / dt
+  //       first_term = u
+  //      second_term = 2 * u_old
+  //       third_term = u_older
+  u_dotdot -= u_old;
+  u_dotdot -= u_old;
+  u_dotdot += u_older;
+  u_dotdot *= 1.0 / (_dt * _dt);
+}
```

Comment:
How does this even work not in a header file?

Reply:
Why would implementation need to be in the header file? I moved it because that seemed wrong to me. But I can move it back if there was a reason.

Reply:
See this answer: https://stackoverflow.com/a/495056

Reply:
I see. Then yes I'm surprised it still works. I'll move it back in my next push.

Reply:
well it works probably because people never instantiate the template types to be anything other than NumericVector (or whatever that is). If that can be guaranteed, you can remove the template and separate the declaration and implementation. Otherwise, I suggest putting it back to the header.